editors/code/src/installation/interfaces.ts

```diff
@@ -0,0 +1,26 @@
+export interface GithubRepo {
+    name: string;
+    owner: string;
+}
+
+export interface ArtifactMetadata {
+    releaseName: string;
+    downloadUrl: string;
+}
+
+
+export enum BinarySourceType { ExplicitPath, GithubBinary }
+
+export type BinarySource = EplicitPathSource | GithubBinarySource;
```

Comment:
This is what you call tagged unions (or enums in Rust).

Reply:
Nit: `Explicit`